src/snowflake/snowpark/modin/plugin/_internal/generator_utils.py

```diff
@@ -225,10 +225,11 @@ def generate_irregular_range(
 
     # TODO: SNOW-1646883 fix invalid identifier error when sql_simplifier_enabled is True
     session = get_active_session()
-    sql_simplifier_enabled = session.sql_simplifier_enabled
-    session.sql_simplifier_enabled = False
-    num_offsets = session.range(start=0, end=periods, step=1)
-    session.sql_simplifier_enabled = sql_simplifier_enabled
+    with session._lock:
+        sql_simplifier_enabled = session.sql_simplifier_enabled
+        session.sql_simplifier_enabled = False
```

Comment:
hmm, I thought we are putting the lock on the sql_simplfier_enable function, if we do that, do we still need to do the change here?

Reply:
this is done so that the value stays constant during `session.range` invocation. This was the intention of the old code implementation and lock just holds the same guarantee.

Reply:
that code here is hacky and bad. do you know what is the ticket SNOW-1646883 referring for, i think we should fix that and remove this code, we are not suppose to change the config in our code like this.

Reply:
I'll fix this bug and remove the hack

Reply:
seems like it is fixed: I removed the hack here: https://github.com/snowflakedb/snowpark-python/pull/2375